code/datums/components/embeds/embedded_structure.dm

```diff
@@ -0,0 +1,67 @@
+/datum/component/embedded_structure
```

Comment:
You should only need a single component for the various types

Reply:
Woops, I just have them separate right now for testing purposes, hence the last commit name